src/dos/cdrom.h

```diff
@@ -107,18 +110,75 @@ class CDROM_Interface
 	virtual bool GetMediaTrayStatus (bool& mediaPresent, bool& mediaChanged, bool& trayOpen) = 0;
 	virtual bool PlayAudioSector    (const uint32_t start, uint32_t len) = 0;
 	virtual bool PauseAudio         (bool resume) = 0;
-	virtual bool StopAudio          (void) = 0;
+	virtual bool StopAudio          () = 0;
 	virtual void ChannelControl     (TCtrl ctrl) = 0;
 	virtual bool ReadSectors        (PhysPt buffer, const bool raw, const uint32_t sector, const uint16_t num) = 0;
 	virtual bool ReadSectorsHost    (void* buffer, bool raw, unsigned long sector, unsigned long num) = 0;
 	virtual bool LoadUnloadMedia    (bool unload) = 0;
-	virtual void InitNewMedia       (void) {}
+	virtual void InitNewMedia       () {}
+};
+
+class CDROM_Interface_SDL : public CDROM_Interface
+{
+public:
+	CDROM_Interface_SDL();
+	CDROM_Interface_SDL(const CDROM_Interface_SDL&);
+	CDROM_Interface_SDL& operator = (const CDROM_Interface_SDL&);
+	virtual ~CDROM_Interface_SDL();
+	virtual bool SetDevice(char *path, int forceCD);
+	virtual bool GetUPC(unsigned char &attr, char *upc)
+	{
+		attr = '\0';
+		strcpy(upc, "UPC");
+		return true;
+	}
+	virtual bool GetAudioTracks(uint8_t &stTrack, uint8_t &end, TMSF &leadOut);
+	virtual bool GetAudioTrackInfo(uint8_t track, TMSF &start, unsigned char &attr);
+	virtual bool GetAudioSub(unsigned char &attr,
+	                         unsigned char &track,
+	                         unsigned char &index,
+	                         TMSF &relPos,
+	                         TMSF &absPos);
+	virtual bool GetAudioStatus(bool &playing, bool &pause);
+	virtual bool GetMediaTrayStatus(bool &mediaPresent,
+	                                bool &mediaChanged,
+	                                bool &trayOpen);
+	virtual bool PlayAudioSector(const uint32_t start, uint32_t len);
+	virtual bool PauseAudio(bool resume);
+	virtual bool StopAudio();
+	virtual void ChannelControl([[maybe_unused]] TCtrl ctrl)
+	{
+		return;
+	}
+	virtual bool ReadSectors([[maybe_unused]] PhysPt buffer,
+	                         [[maybe_unused]] const bool raw,
+	                         [[maybe_unused]] const uint32_t sector,
+	                         [[maybe_unused]] const uint16_t num)
+	{
+		return false;
+	}
+	virtual bool ReadSectorsHost([[maybe_unused]] void *buffer,
+	                             [[maybe_unused]] bool raw,
+	                             [[maybe_unused]] unsigned long sector,
+	                             [[maybe_unused]] unsigned long num)
+	{
+		return true;
+	}
+	virtual bool LoadUnloadMedia(bool unload);
+
+private:
+	bool Open();
+	void Close();
+
+	SDL_CD *cd = nullptr;
+	int driveID = 0;
+	Uint32 oldLeadOut = 0;
 };
 
 class CDROM_Interface_Fake final : public CDROM_Interface
 {
 public:
-	bool SetDevice          (char *) { return true; }
+	bool SetDevice          (char *, [[maybe_unused]] int forceCD) { return true; }
```

Comment:
I just realized that I'm not really understanding the intent of "`forceCD`".
My confusion:
 - "Forcing" or "not-forcing" sound boolean to mean, but now I see this is an integer.
 - Force 1, Force 5, Force 10: I'm not sure what the force numbers mean?
 - the entire header deals with CDROM "stuff", so forcing-CD sounds almost meaningless.

Is forceCD a physical CDROM drive identifier? (that's my best guess.. only because of the nature of this PR.  Maybe it could be mass-replaced in this PR to a better name)?

Reply:
I believe it is intended to be the CD drive number. Changed forceCD to cd_number accordingly.